### DISCUSSION TITLE
Azure Service Bus

### DISCUSSION THREAD

**Discussion by ItsRobbAllen** [ravendb/ravendb#17449] [General]
I saw the 6.0 release email today & it mentioned Kafka & Rabbit MQ which got me thinking about my situation where we use Azure Service Bus.

I realize Raven can't write plugins for every conceivable external system, but is there any way to write a custom plug-in for a different message queue and deploy it to Raven Cloud for our systems?

I'd LOVE to skip having a subscription simply to convert a document to a queue message, but alas we use AzSB and not Kafka or Rabbit :/

**Comment by ayende** [comment]
We would love to have support for Azure Service Bus.
Now that we built the infrastructure, the addition on new sinks should be straight forward, see:
https://github.com/ravendb/ravendb/tree/v6.0/src/Raven.Server/Documents/QueueSink

Is this something that you can do a pull request for?

**Reply by ItsRobbAllen** [reply]
Is this valid for 5.x?

**Reply by ppekrol** [reply]
The infrastructure was done for 6.0, in 5.x it would have to be done from scratch.

**Reply by ItsRobbAllen** [reply]
OK then, I'll just wait until we're ready to upgrade to 6.0.
